pkg/network/tracer/tracer_test.go

```diff
@@ -1069,14 +1072,22 @@ func (s *TracerSuite) TestDNSStats() {
 	t := s.T()
 	cfg := testConfig()
 	cfg.CollectDNSStats = true
-	cfg.DNSTimeout = 1 * time.Second
+	cfg.DNSTimeout = 5 * time.Second
 	tr := setupTracer(t, cfg)
 	t.Run("valid domain", func(t *testing.T) {
 		testDNSStats(t, tr, "golang.org", 1, 0, 0, validDNSServer)
 	})
 	t.Run("invalid domain", func(t *testing.T) {
 		testDNSStats(t, tr, "abcdedfg", 0, 1, 0, validDNSServer)
 	})
+}
+
+func (s *TracerSuite) TestDNSStatsWithTimeout() {
+	t := s.T()
+	cfg := testConfig()
+	cfg.CollectDNSStats = true
+	cfg.DNSTimeout = 1 * time.Second
```

Comment:
Can we try `0` here?

Reply:
apparently not:
`suite.go:87: test panicked: non-positive interval for NewTicker`

Reply:
I can set it to 1ms?

Reply:
Lets set it to `250ms`, that ticker should fire before we do the asserts and set the timeout; the test has a sleep for `500ms` so this should fire before that.